build.gradle

```diff
@@ -42,8 +42,35 @@ if (!repositories) {
   print("This machine already has some pre-configured repositories, will use: ${repositories*.name.join(", ")}")
 }
 
+buildscript {
+  dependencies {
+    if (JavaVersion.current().isJava11Compatible()) {
+      classpath("com.autonomousapps:dependency-analysis-gradle-plugin:$dependencyAnalysisPluginVersion")
+    }
+  }
+}
+
 apply plugin: "io.servicetalk.servicetalk-gradle-plugin-internal-root"
 
+// Unfortunately, we can not make this validation part of our servicetalk-gradle-plugin-internal because it requires
+// JDK11 while we build and package out plugin with JDK8.
+if (JavaVersion.current().isJava11Compatible()) {
+  apply plugin: "com.autonomousapps.dependency-analysis"
+
+  dependencyAnalysis {
+    issues {
+      all {
+        onAny {
+          severity("warn")  // FIXME: switch to "fail" when all issues are addressed
```

Comment:
This will change after all issues are resolved, will require a few follow-up PRs